homeassistant/components/poolstation/entity.py

```diff
@@ -0,0 +1,37 @@
+"""Base class for Poolstation entity."""
+from __future__ import annotations
+
+from pypoolstation import Pool
+
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PoolstationDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class PoolEntity(CoordinatorEntity):
+    """Representation of a pool entity."""
+
+    coordinator: PoolstationDataUpdateCoordinator
+
+    def __init__(
+        self,
+        pool: Pool,
+        coordinator: PoolstationDataUpdateCoordinator,
+        entity_suffix: str,
+    ) -> None:
+        """Init from config, hookup pool and coordinator."""
+        super().__init__(coordinator)
+        self._pool = pool
+
+        pool_id = self._pool.id
+        name = self._pool.alias
+
+        self._attr_name = f"{name}{entity_suffix}"
+        self._attr_unique_id = f"{name}{entity_suffix}"
```

Comment:
Can the alias be changed by the user? If yes, don't use it as part of unique_id.

Reply:
It can be changed, however it is unique within you own list of pools. There is a stable `id` i can use but that means sensor names will be quite unpleasant to use, right? Think `sensor.my_pool_current_ph` vs `sensor.9123_current_ph`. I am open to change it if you deem that the annoyance of leaving orphan entities if you rename your pool in the platform (and have to manually delete them) is more important han generating friendlier entity names.

Reply:
If you rename the pool, Home Assistant will no longer recognize the entity and create a new one because the unique_ids are different.

> There is a stable id i can use but that means sensor names will be quite unpleasant to use, right?

The `entity_id` is visible to the user and generated based on the entity platform and name: `sensor.my_pool_current_ph`.
The `unique_id` is not visible to the user and must always remain the same: `sensor.9123_current_ph`.

If the entity does not have a `unique_id`, the `entity_id` will be used as `unique_id`. You can't change the `entity_id` in the UI if the entity doesn't have a `unique_id`.

Reply:
You are mixing up unique_id and entity_id. The former shall not be user changeable, but will never be seen by a user. See https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

The entity_id is the one like `sensor.my_pool_ph` and gets created by slugifying the `name` property. 

Thanks to the unique_id the entity_id can be changed by a user. Entity_id is what is used by automations etc.

Reply:
Got it. I changed it so `_attr_name` uses the alias, as it's more friendly towards users, but the `_attr_unique_id ` uses the id to make sure it's truly unique and never changes.